integrations/axum/src/lib.rs

```diff
@@ -644,17 +644,21 @@ where
     );
 
     move |req| {
-        let uri = req.uri();
         // 1. Process route to match the values in routeListing
-        let path = uri.path();
+        let path = req
+            .extensions()
+            .get::<MatchedPath>()
+            .expect("Failed to get Axum router rule")
+            .as_str();
         // 2. Find RouteListing in paths. This should probably be optimized, we probably don't want to
         // search for this every time
         let listing: &RouteListing =
-            paths.iter().find(|r| r.path() == path).expect(
-                "Failed to find the route {path} requested by the user. This \
+            paths.iter().find(|r| r.path() == path).expect(&format!(
+                "Failed to find the route {} requested by the user. This \
                  suggests that the routing rules in the Router that call this \
                  handler needs to be edited!",
-            );
+                path
```

Comment:
Why was this moved out of the format string I'm curious?

Reply:
Ah, because the original was not a format string, and was just saying {path} instead of a useful error message containing the route that was broken

Reply:
Also, not from your code but just a prior-existing issue, `format` should not not not be used in `expect`, that means it's doing string formatting and allocation every single call, even when the expect is not being done, instead it should be in a closure, like via `.unwrap_or_else(|_| panic!("Failed to find the route {path} requested by the user. This suggests that the routing rules in the Router that call this handler needs to be edited!"))` that way the formatting operation isn't done unless it actually needs to be done

Reply:
> Ah, because the original was not a format string, and was just saying {path} instead of a useful error message containing the route that failed to match

Uh, it shouldn't?  `format!("Blah: {path}")` will return a string of `Blah: /blorp` if path is `/blorp`.

Reply:
Ah I misread, you added the format call, yeah format definitely doesn't go in an expect, use the unwrap_or_else I put above.  :-)